ert/data/_record.py

```diff
@@ -141,46 +144,77 @@ class BlobRecord(Record):
 
 class RecordCollection(_DataElement):
     records: Union[Tuple[NumericalRecord, ...], Tuple[BlobRecord, ...]]
-    ensemble_size: Optional[int] = None
+
+    # Records with a invariant content can be represented with a single record,
+    # broadcasted to a given ensemble size. Setting the invariant field makes a
+    # collection invariant, with the given size. If no ensemble size is given,
+    # it is set to 1.
+    invariant: bool = False
```

Comment:
Is this really needed?

Why not just decouple the check for consistent ensemble size the resould would look something like this:
```
class RecordCollection(_DataElement):
    records: Union[Tuple[NumericalRecord, ...], Tuple[BlobRecord, ...]]
    ensemble_size: Optional[int] = None

    @property
    def record_type(self) -> Optional[RecordType]:
        return self.records[0].record_type

    @validator("ensemble_size", pre=True, always=True)
    def ensemble_size_validator(
        cls, ensemble_size: Optional[int], values: Dict[str, Any]
    ) -> Optional[int]:
        if ensemble_size is None and "records" in values:
            ensemble_size = len(values["records"])
        assert ensemble_size is not None and ensemble_size > 0
        return ensemble_size

    @root_validator(skip_on_failure=True)
    def ensure_consistent_ensemble(
        cls, values: Dict[str, Any]
    ) -> Dict[str, Any]:
        assert "records" in values and "ensemble_size" in values
        record_type = values["records"][0].record_type
        for record in values["records"][1:]:
            if record.record_type != record_type:
                raise ValueError("Ensemble records must have a uniform record type")
        if len(values["records"]) == 1 and len(values["records"]) != values["ensemble_size"]:
            values["records"] = values["records"] * values["ensemble_size"]
        return values
```

Reply:
When you loose the `invariant` field and duplicate the values like this, you cannot afterwards track easily if the collection is invariant. The `invariant` field is used elsewhere in the code. 

The check if the size is not equal to the ensemble size does not work if the ensemble size is equal to one, and you want to mark the collection as invariant. Which we need when we load a blob record, unless we stick to the limitation that blob records are always implicitly invariant, and numerical records not.

You could use the duplication, and stil store an `invariant` field, I suppose, but I do not find that very elegant.